### PR TITLE
Demo third-party links with real external web site

### PR DESCRIPTION
**Pull request by smattingly**
You can test this at https://dewv.net/ftm/test.html, where I have set up a demo of a mechanism for embedded third party links.

(Note second magnifying glass icon is temporarily used for the third party link.)
I think this approach is simple and reasonably abuse-proof.
For any and only sites that we put in the partner list, their embed requests will be served with links in the embed location list. Those site(s) will just need to handle `/ftm?id=X` requests.

### CODE REVIEW

**Comment by awong-dev** [comment]
Interesting. This looks pretty good!

So basically each specific referrer would need to submit a PR to have themselves added with their own icon.

And then the callback URL format is also similarly encoded (nothing dynamic via parameters).

I like it!

File: applicationRoutes.js [hunk 1]
```diff
@@ -43,7 +43,17 @@ router.get('/faq', (req, res) => {
   });
 });
 
+const linkPartners = [
```
Comment (awong-dev) [inline applicationRoutes.js:46]: Can we pull out the various bits where we'd want a partner to change stuff (so the referer allow-list and the icon) into a separate file so it's very easy to know it's 3rd party config？

File: client/locations-list-map.js [hunk 2]
```diff
@@ -1090,6 +1096,9 @@ function getEntryEl(entry) {
     sendEvent('listView', 'clickZoom', entry.name);
     zoomToMarker(entry.marker);
   });
+  $(entry.domElem).find('.entry-partner-link').on('click', () => {
+    window.open(`${document.body.dataset.partnerSite}ftm?id=${entry.row}`, '_blank');
```
Comment (awong-dev) [inline client/locations-list-map.js:1100]: Is it worth having some config allow for partners to declare their own callback URL?  Or is that oo much for now?

**Comment by awong-dev** [comment]
Tagging @ngottlieb since we were chatting.  This approach seems to meet the security requirements and be even a little bit simpler (ie no URL parameter.)

File: client/locations-list-map.js (hunk 2 above)
Comment (ngottlieb) [reply client/locations-list-map.js:1100]: we could just make the `linkPartners` from `applicationRoutes.js` (abstracted to another file) into an object keyed by the referrer, values being the callback URL?

File: client/locations-list-map.js [hunk 3]
```diff
@@ -952,10 +952,16 @@ function createRequesterListItemEl(entry) {
...
   const header = ce('div', 'd-flex');
   const headerHospitalInfo = ce('div', 'flex-grow-1');
+  const headerOrgType = ce('div', 'flex-grow-1 d-flex justify-content-end text-pink');
   const headerZoomLink = ce('div', 'icon icon-search entry-zoom-link');
   headerZoomLink.setAttribute('aria-label', 'Zoom to marker');
-  const headerOrgType = ce('div', 'flex-grow-1 d-flex justify-content-end text-pink');
-  ac(headerHospitalInfo, ce('h5', null, [ctn(entry.name), headerZoomLink]));
+  const children = [ctn(entry.name), headerZoomLink];
+  if (document.body.dataset.partnerSite) {
+    const headerPartnerLink = ce('div', 'icon icon-search entry-partner-link'); // TODO replace icon
```
Comment (ngottlieb) [inline client/locations-list-map.js:960]: If we canonically convert `dataset.partnerSite` into a CSS-class-safe string, then all our partners would have to do is add the CSS for their icon (and add themselves to the whitelist in our routing)

File: applicationRoutes.js (hunk 1 above)
Comment (smattingly) [reply applicationRoutes.js:46]: I think that's a good idea.

File: client/locations-list-map.js (hunk 2 above)
Comment (smattingly) [reply client/locations-list-map.js:1100]: As @ngottlieb says, it wouldn't be hard to do. However, the original request as it filtered to me sounded maybe like partners were saying "Can you tell us how to ... ?".  You all know the partners and I don't-- your call on whether simple/standard or minor complexity/flexible is better. Or both: allow custom callback but default to `/ftm?id=X` when not specified.

**Comment by smattingly** [comment]
Closing this PR without merge so I can create another to get a redeploy.

**Pull request closed without merging**